app/controllers/offences_controller.rb

```diff
@@ -13,23 +13,19 @@ class OffencesController < ApplicationController
   skip_load_and_authorize_resource only: [:index]
 
   def index
-    if permitted_params[:fee_scheme].present?
-      @offences = FeeReform::SearchOffences.call(permitted_params)
-
-      respond_to do |format|
-        format.json do
-          render json: @offences, each_serializer: FeeReform::OffenceSerializer
-        end
-      end
-    else
-      @offences = Offence.in_scheme_nine.where(offence_filter)
-      @data = @offences.map { |offence| { offence_id: offence.id, id: offence.offence_class.id, description: offence.offence_class.description } }
-      render json: @data
+    respond_to do |format|
+      format.json { render json: offences, each_serializer: FeeReform::OffenceSerializer }
     end
   end
 
   private
 
+  def offences
+    return FeeReform::SearchOffences.call(permitted_params) if permitted_params[:fee_scheme].present?
+
+    Offence.in_scheme_nine.where(offence_filter)
```

Comment:
a mapping has been removed for the non-fee-scheme flow. is it handled elsewhere?
```
      @data = @offences.map { |offence| { offence_id: offence.id, id: offence.offence_class.id, description: offence.offence_class.description } }
      render json: @data
```

Reply:
That mapping was added for me to test if I was getting the right data from the controller. These two lines does not exists in master